src/test/java/com/github/cameltooling/lsp/internal/hover/KameletHoverTest.java

```diff
@@ -18,47 +18,45 @@
 
 import static org.assertj.core.api.Assertions.assertThat;
 
+import java.net.URISyntaxException;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Stream;
 
 import org.eclipse.lsp4j.Hover;
 import org.eclipse.lsp4j.HoverParams;
 import org.eclipse.lsp4j.Position;
 import org.eclipse.lsp4j.TextDocumentIdentifier;
-import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
 
 import com.github.cameltooling.lsp.internal.AbstractCamelLanguageServerTest;
 import com.github.cameltooling.lsp.internal.CamelLanguageServer;
 
 class KameletHoverTest extends AbstractCamelLanguageServerTest {
 
-	@Test
-	void testProvideDocumentationOnHover() throws Exception {
-		CamelLanguageServer camelLanguageServer = initializeLanguageServer("<from uri=\"kamelet:aws-ddb-streams-source\" xmlns=\"http://camel.apache.org/schema/spring\"></from>\n");
+	@ParameterizedTest(name = "{0}")
+	@MethodSource
+	void testHoverDocumentation(String testName, String camelUri, int characterPositionInCamelUriToTest, String expectedHoverContent)
+			throws URISyntaxException, InterruptedException, ExecutionException {
+		String beforeCamelUriToTest = "<from uri=\"";
```

Comment:
Might be good to add this as a constant to AbstractCamelLanguageServerTest, since it seems like it might be generic enough to be used in many places. Just a thought though.

Reply:
definitely something that can be done to improve it. I created https://issues.redhat.com/browse/FUSETOOLS2-1166